vnext/local-cli/runWindows/utils/commandWithProgress.js

```diff
@@ -50,18 +71,23 @@ function commandWithProgress(spinner, taskDoingName, command, args, verbose) {
     }
 
     const cp = child_process.spawn(command, args, spawnOptions);
-
+    var error = null;
     if (!verbose) {
       cp.stdout.on('data', chunk => {
         const text = chunk.toString();
         setSpinnerText(spinner, taskDoingName + ': ', text);
       });
       cp.stderr.on('data', chunk => {
         const text = chunk.toString();
+        if (!error) {
+          error = text;
+        }
         if (verbose) {
           console.error(chalk.red(text));
         }
-        setSpinnerText(spinner, taskDoingName + ': ERROR: ', text);
+        if (text) {
+          setSpinnerText(spinner, taskDoingName + ': ERROR: ', error);
+        }
```

Comment:
Should we reject the promise if we see stderr output? Could see either way being valid.

Reply:
I think as is this would also repeatedly rewrite the spinner text as new stderr lines come in, so likely flashing then showing the last. That doesn't seem like the right behavior.

Reply:
up until now it was flashing with each line coming in. This fixes things so that only the first error line for a task is ever printed.

---
In reply to: [405941744](https://github.com/microsoft/react-native-windows/pull/4536#discussion_r405941744) [](ancestors = 405941744)

Reply:
Since this helper is being used outside of just the powershell scenario, I'd rather keep that part of the semantics the same. Happy to be convinced otherwise if folks feel strongly about it.

---
In reply to: [405940387](https://github.com/microsoft/react-native-windows/pull/4536#discussion_r405940387) [](ancestors = 405940387)

Reply:
Oh, I think I get this logic now. We set the spinner to first error bc we mutate `error` before subsequent calls and output everything if verbose.

I still think rejecting the promise might be appropriate if we see stderr though, especially if we mark the spinner as failed.

Reply:
@asklar The reason I lean towards rejection is what the UX would look like. We'd show a command fail, then chug on? Seems like that could potentially do the wrong thing.

Reply:
@ngerlem@microsoft.com  Sadly not all programs use stderr for errors. For example MIDL uses stderr for some messages it wants to print out to the console:


```
C:\rnw>midl 1>NUL
Microsoft (R) 32b/64b MIDL Compiler Version 8.01.0622
Copyright (c) Microsoft Corporation. All rights reserved.
```

---
In reply to: [405944555](https://github.com/microsoft/react-native-windows/pull/4536#discussion_r405944555) [](ancestors = 405944555)

Reply:
Oof. Yeah, that makes sense then.

Reply:
Wait, if midl outputs to stderr. Wouldn't that make us always show errors and show the spinner as failed?

Reply:
the task still succeeds even if the spinner temporarily shows an error. It's like we go "OH THERE'S AN ERROR... nevermind all good, fam"

---
In reply to: [405950409](https://github.com/microsoft/react-native-windows/pull/4536#discussion_r405950409) [](ancestors = 405950409)